src/lib/Sympa/Message.pm

```diff
@@ -2145,6 +2152,7 @@ sub _urlize_one_part {
         $filename = Encode::encode_utf8($filename)
             if Encode::is_utf8($filename);
     } else {
+        return undef if ($entity->effective_type =~ m{\A(multipart|text)});
```

Comment:
Attachment with text/plain type should be urlized. Nested multipart should also urlized.

Reply:
I'm not sure actually. But I might have a biased opinion.
On my opinion, the main point of urlized is to decrease the size of mails sent. Consequently, I want to get rid of images, PDF and all theses attachments that are heavy.
Text is rarely that heavy. So if we keep all mulitpart and text parts, but get rid of the rest, we have really lighter messages which is good enough for me, as we would keep all the text that gives sens to the message.
However, I might have misunderstood the way _urlize_one_part works and it may ignore the content of nested multipart. In that case, this commit needs reworking. I'll add some tests about this situation.

Reply:
OK. Inded, if I ignore a multipart, it won't go recursively inside it to extract attachments.
So I think I need to treat "multipart/mixed" differently.

Reply:
I refactored the code slightly to allow recursive urlization of nested multiparts.